### PR TITLE
Task 4 from threads is done.

### PR DESCRIPTION
**Pull request by qnislavnachev**
Here we have 2 threads which 1 of them has to add item to a  list and the other one has to remove last item, when list is full adding thread has to wait to be removed item from list, and when list is empty removing thread has to wait to be added item to the list.

### CODE REVIEW

File: src/main/java/task4/MyList.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public final List<Integer> newList;
+    public final int maxSize;
+
+    public MyList(int maxSize) {
+        this.maxSize = maxSize;
+        newList = new ArrayList<Integer>(maxSize);
+    }
+
+    public synchronized void insert(int number) {
+        while (maxSize == newList.size()) {
```
Comment (tsony-tsonev) [inline src/main/java/task4/MyList.java]: why we need this while

Comment (qnislavnachev) [inline src/main/java/task4/MyList.java]: Now I see we dont need this while when we `ve got the while pool in run method.

Comment (qnislavnachev) [inline src/main/java/task4/MyList.java]: fixed

File: src/main/java/task4/MyList.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public synchronized void remove() {
+        if (newList.isEmpty()) {
+            try {
+                wait();
+            } catch (InterruptedException e) {
+                System.out.println("It was interrupted.");
+            }
+        }
+        System.out.println("Last item is removed!");
+        newList.remove(newList.size() - 1);
```
Comment (tsony-tsonev) [inline src/main/java/task4/MyList.java]: you should be able to remove all the elements from the list not just the last one

File: src/main/java/task4/RemovingThread.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    public final MyList list;
+    private int count = 0;
+
+    public RemovingThread(MyList list) {
+        this.list = list;
+    }
+
+    @Override
+    public void run() {
+        while (count <= 10) {
```
Comment (tsony-tsonev) [inline src/main/java/task4/RemovingThread.java]: if you want to limit the removal attempts do it from outside - pass the attempts count to the thread

File: src/main/java/task4/RemovingThread.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            try {
+                sleep(5000);
```
Comment (tsony-tsonev) [inline src/main/java/task4/RemovingThread.java]: why do we need this sleep?

Comment (qnislavnachev) [inline src/main/java/task4/RemovingThread.java]: I am using this sleep to check my program if it work correctly

File: src/main/java/task4/MyList.java (hunk 2 above)
Comment (qnislavnachev) [inline src/main/java/task4/MyList.java]: fixed

File: src/main/java/task4/RemovingThread.java (hunk 3 above)
Comment (qnislavnachev) [inline src/main/java/task4/RemovingThread.java]: fixed

File: src/main/java/task4/RemovingThread.java (hunk 4 above)
Comment (qnislavnachev) [inline src/main/java/task4/RemovingThread.java]: fixed

File: src/main/java/task4/Demo.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package task4;
+
+public class Demo {
+    public static void main(String[] args) {
+        MyList list = new MyList(5);
+        AddingThread add = new AddingThread(list, 10);
+        RemovingThread remove = new RemovingThread(list, 10, 0);
```
Comment (tsony-tsonev) [inline src/main/java/task4/Demo.java]: Now you remove only the first element from the list. This index handling should be done in the implementation of the list, you just need to increment an index when you insert, when you remove you remove by that index and then you decrement it.

Comment (qnislavnachev) [inline src/main/java/task4/Demo.java]: fixed

**Comment by tsony-tsonev** [comment]
LGTM

**Comment by qnislavnachev** [comment]
This pull request will be closed because It`s  compared with master branch

**Pull request closed without merging**